[FIX] base: Spanish states

pedrobaeza opened the pull request:
According official country source (BOE) https://www.boe.es/diario_boe/txt.php?id=BOE-A-2011-11606, some states changed their denomination to allocate local language spelling.

As there's a lot of people that still uses the old one, the best solution is to include both words in the name of the state, and thus user can search by any of them. This is the same technique that we use in our localization module l10n_es_toponyms. As v10 now includes this list, we will remove the list from the module, but it's important to have it correct here.

@mart-e I think this one is for you

pedrobaeza commented:
cc @Tecnativa

pedrobaeza commented:
I have included also a state code this is incorrect according followed code convention. You can squash both commits if wanted.

mart-e commented on odoo/addons/base/res/res.country.state.csv:
```diff
@@ -414,43 +414,43 @@ state_it_vr,it,"Verona","VR"
+state_es_vi,es,"Araba/Álava","VI"
 state_es_ab,es,"Albacete","AB"
-state_es_a,es,"Alicante","A"
+state_es_a,es,"Alicante / Alacant1","A"
```
Is the 1 at the end expected?

pedrobaeza replied:
Of course not. Typo approaching!

pedrobaeza commented:
I'm using GitHub interface, so you should squash all the possible commits.

mart-e commented on odoo/addons/base/res/res.country.state.csv:
```diff
@@ -414,43 +414,43 @@ state_it_vr,it,"Verona","VR"
 state_es_m,es,"Madrid","M"
 state_es_ma,es,"Málaga","MA"
-state_es_ml,es,"Melilla","ML"
+state_es_me,es,"Melilla","ME"
```
the change of external id can not be made in stable as it would mean it is a new state (and deleting the previous entries)

pedrobaeza replied:
Well, I think we can prepare a migration script for changing it. If you don't update, no problem. If you update, no problem also :wink: If not, we have the same problem with the migration path to v11, but that problem can also be handled in the migration scripts v10 > v11. What you prefer.

KangOl replied:
I would prefer **not** having this change made in stable version.

The rename will be handled in a migration script for v11 (actually next saas version)

mart-e commented:
No problem to squash.
Also wondering, sometimes you put the new name before, sometimes after. It could be good to be consistent (like always the new name first)

pedrobaeza commented:
Well, the inconsistency comes from "Álava", which is officially renamed to "Araba/Álava" (both names, first regional name and then Spanish one, without spaces).

I have removed then the first part from "Álava / Araba/Álava", for not being redundant. The rest are formed by "Spanish / Regional" names. What do you think of the best ordering rule and handling this special case)?

mart-e commented:
@pedrobaeza and what about  `<new name> (<old>)`?

pedrobaeza commented:
OK, I have switched to traditional git to get rid of the branch and squash commits.

@mart-e, I have changed the states names according your suggestion and indeed looks better.

@KangOl, are you handling then the migration of the XML-ID or should I make any PR to saas branch?

pedrobaeza commented:
Thanks!

mart-e commented:
Thanks for the update, merged in 10 at https://github.com/odoo/odoo/commit/912dd834038567bf0fc899e3720637f14f3c5897

If you wish to make a master (or saas-14) pull request with xmlid change, I am sure @KangOl will be glad :)

pedrobaeza commented:
I see that the change it's still not propagated to saas-14, so I need to wait or my PR will have conflicts. What's the frequency for merging changes?

mart-e commented:
It is usually done around twice a week depending on the branch. You can anyway create a patch changing just this line and we will see it at forward port where the lines conflicts.

KangOl commented:
The forward-port has been made.

pedrobaeza commented:
Thanks, I'll prepare the PR for saas-14 branch then.

tivisse commented:
@pedrobaeza Thanks for the feedback.

Your issue has been fixed in 8.0 at https://github.com/odoo/odoo/commit/
